driver/src/main/java/org/kaazing/k3po/driver/internal/control/handler/ControlServerHandler.java

```diff
@@ -106,6 +109,28 @@ public void prepareReceived(final ChannelHandlerContext ctx, MessageEvent evt) t
             return;
         }
 
+        if ((lastRobot != null && ! lastRobot.getDisposedFuture().isDone()) || ! testFinished.compareAndSet(true, false)) {
+            // we need to wait for it to end
+            lastRobot.getDisposedFuture().addListener(new ChannelFutureListener() {
+                @Override
+                public void operationComplete(ChannelFuture future) throws Exception {
+                    ((NioSocketChannel)ctx.getChannel()).getWorker().executeInIoThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            try {
+                                prepareReceived(ctx, evt);
+                            } catch (Exception e) {
+                                sendErrorMessage(ctx, e);
```

Comment:
I have tried to see how the old k3po handles an exception in prepareReceived. It will not respond with any error message and the test times out. So I included everything in a try-catch and removed the `throws Exception` statement. So this piece of code is not required anymore. Please see changes.